src/main/java/in/ac/bits/protocolanalyzer/analyzer/EventBusFactory.java

```diff
@@ -18,7 +18,7 @@
 
 @Service
 @Configurable
-class EventBusFactory {
+public class EventBusFactory {
```

Comment:
any reason for making this class public?

Reply:
I was not able to create and inject a bean for the mentiones class unless it was public.

Reply:
As per the [access control](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html) page, there is a difference between no access modifier (package private) and protected. The existing declaration with no access modifier would not allow imports of the class in another package. But with protected access specifier, we can import the class in other packages.

Reply:
@adityaketkar please try protected access specifier. If it works, do retain it.

Reply:
@prasadtalasila I think non-inner classes cannot be declared as protected. Also, "protected" gives extended access to child classes, but tests do not inherit the EventBusFactory class

Reply:
@adityaketkar you are right. Lets go with public access specifier.